nixos/modules/services/databases/postgresql.nix

```diff
@@ -36,6 +36,32 @@ let
 
   groupAccessAvailable = versionAtLeast postgresql.version "11.0";
 
+  upgradeScript =
+    assert lib.versionOlder cfg.upgradeFrom.package.version cfg.package.version;
+    pkgs.writeShellApplication {
+      name = "upgrade.sh";
+      runtimeInputs = [];
+      text = ''
+        # There is no old version to upgrade from - noop
+        [[ -d "${cfg.upgradeFrom.dataDir}" ]] || exit 0
```

Comment:
We should at least log something. If you wrote the wrong postgres version in the settings you will wonder why nothing happens.

Reply:
That would log on every start of postgresql, when you cleanup the old data dir though 🤔 
(also in the case there never was an older version deployed, e.g. clean deploy directly to newer version)

Reply:
I think once you've done it once, you may want to be reminded to remove a stale `upgradeFrom`. What do you think?

Reply:
Added a log.